### PR TITLE
Annual cleanup and warning fix

### PR DESCRIPTION
**Pull request by NHOrus**
Updated LodePNG version, did some things to enums and readability in AIModule

### CODE REVIEW

File: src/Battlescape/TileEngine.cpp [hunk 1]
```diff
@@ -1588,7 +1588,7 @@ int TileEngine::verticalBlockage(Tile *startTile, Tile *endTile, ItemDamageType
 	int block = 0;
 
 	// safety check
-	if (startTile == 0 || endTile == 0) return 0;
+	if (!startTile || !endTile == 0) return 0;
```
Comment (SupSuper) [inline src/Battlescape/TileEngine.cpp]: This is wrong

File: src/Interface/NumberText.cpp [hunk 2]
```diff
@@ -183,11 +183,11 @@ NumberText::NumberText(int width, int height, int x, int y) : Surface(width, hei
 		_borderedChars[i] = new Surface(5, 7);
 		// give it a border
 		// this is the "darker" shade that goes in the corners.
-		for (int x = 0; x <= 2; x += 2)
```
Comment (SupSuper) [inline src/Interface/NumberText.cpp]: Would be better to keep the "x" and "y" since they refer to coordinates. Maybe "xx" and "yy"?

**Comment by SupSuper** [comment]
In general:
- Best not to touch the null pointer checks, wouldn't wanna cause an error and accidentally introduce a subtle bug.
- When renaming variables due to "shadowing", make sure the new names are still meaningful.

File: src/Battlescape/Pathfinding.cpp [hunk 3]
```diff
@@ -668,13 +668,14 @@ bool Pathfinding::isBlocked(Tile *tile, const int part, BattleUnit *missileTarge
 		}
 	}
 	// missiles can't pathfind through closed doors.
-	if (missileTarget != 0 && tile->getMapData(part) &&
-		(tile->getMapData(part)->isDoor() ||
-		(tile->getMapData(part)->isUFODoor() &&
-		!tile->isUfoDoorOpen(part))))
+	{ TilePart tp = (TilePart)part;
```
Comment (SupSuper) [inline src/Battlescape/Pathfinding.cpp]: Why the extra bracket?

Comment (NHOrus) [reply src/Battlescape/Pathfinding.cpp]: Scoping. So `tp` will go away after closing bracket.

File: src/Battlescape/TileEngine.cpp (hunk 1 above)
Comment (NHOrus) [reply src/Battlescape/TileEngine.cpp]: Fixed by amending commit

File: src/Menu/VideoState.cpp [hunk 4]
```diff
@@ -317,13 +317,13 @@ static introSoundEffect introSoundTrack[] =
...
 {
-	Mod *mod;
+	Mod *_mod;
 	Music *m;
 	Sound *s;
 	int trackPosition;
 	FlcPlayer *_flcPlayer;
 
-	AudioSequence(Mod *mod, FlcPlayer *flcPlayer) : mod(mod), m(0), s(0), trackPosition(0), _flcPlayer(flcPlayer)
+	AudioSequence(Mod *mod, FlcPlayer *flcPlayer) : _mod(mod), m(0), s(0), trackPosition(0), _flcPlayer(flcPlayer)
```
Comment (NHOrus) [inline src/Menu/VideoState.cpp]: This was bad idea, should have gone with `AudioSequence(Mod *_mod, FlcPlayer *flcPlayer)` instead

Comment (NHOrus) [reply src/Menu/VideoState.cpp]: Fixed

**Pull request merged**